hugr-passes/src/replace_types.rs

```diff
@@ -255,19 +272,25 @@ impl TypeTransformer for ReplaceTypes {
     type Err = ReplaceTypesError;
 
     fn apply_custom(&self, ct: &CustomType) -> Result<Option<Type>, Self::Err> {
-        Ok(if let Some(res) = self.type_map.get(ct) {
-            Some(res.clone())
-        } else if let Some(dest_fn) = self.param_types.get(&ct.into()) {
+        let mut ty = None;
```

Comment:
nit: maybe `ty_and_opts` or something similar is clearer (especially since you shadow `ty` below)?